res/controllers/Denon-MC7000-scripts.js

```diff
@@ -0,0 +1,802 @@
+        if (value === 0x01) {
+            engine.setValue(group, "pitch_up", true);
+        } else if (value === 0x7F) {
+            engine.setValue(group, "pitch_down", true);
+        }
+    };
+
+    // Assign Channel to Crossfader
+    MC7000.crossfaderAssign = function(channel, control, value, status, group) {
+        if (value === 0x00) {
+            engine.setValue(group, "orientation", 1); // Centre position
+        } else if (value === 0x01) {
+            engine.setValue(group, "orientation", 0); // Right position
+        } else if (value === 0x02) {
+            engine.setValue(group, "orientation", 2); // Left position
+        }
+    };
+
+    // Assign Spinback length to STOP TIME knob
+    MC7000.stopTime = function(channel, control, value, status, group) {
+        var deckNumber = script.deckFromGroup(group);
+        // "factor" for engine.brake()
+        // this formula produces factors between 31 (min STOP TIME for ca 7 sec back in track)
+        // and 1 (max STOP TIME for ca 18.0 sec back in track)
+        MC7000.factor[deckNumber] = (1.1 - (value / 127)) * 30 - 2;
+    };
+
+    // Use the CENSOR button as Spinback with STOP TIME adjusted length
+    MC7000.censor = function(channel, control, value, status, group) {
+        var deckNumber = script.deckFromGroup(group);
+        var deck = parseInt(group.substring(8, 9)); // work out which deck we are using
+        engine.brake(deck, value > 0, MC7000.factor[deckNumber], - 15); // start at a rate of -15 and decrease by "factor"
+    };
+};
+
+/* SET CROSSFADER CURVE */
+MC7000.crossFaderCurve = function(control, value) {
+    script.crossfaderCurve(value);
+};
+
+/* Set FX wet/dry value */
+MC7000.fxWetDry = function(midichan, control, value, status, group) {
+    var numTicks = (value < 0x64) ? value: (value - 128);
+    var newVal = engine.getValue(group, "mix") + numTicks/64*2;
+    engine.setValue(group, "mix", Math.max(0, Math.min(1, newVal)));
+
+
+};
+
+/* LEDs for VuMeter */
+// VuMeters only for Channel 1-4 / Master is on Hardware
+MC7000.VuMeter = function(value, group) {
+    var VULevelOutValue = engine.getValue(group, "PeakIndicator") ? MC7000.VuMeterLEDPeakValue : value*value*value*value*0x69,
+        deckNumber = script.deckFromGroup(group);
+
+    midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, VULevelOutValue);
+};
+
+/* LEDs around Jog wheel */
+MC7000.JogLed = function(value, group) {
+    var deckNumber = script.deckFromGroup(group);
+    // do nothing before track starts
+    if (value < 0) return;
+
+    var trackDuration = engine.getValue(group, "duration"),
+        position = value * trackDuration / 60 * MC7000.scratchParams.recordSpeed,
+        // LED ring contains 48 segments with each LED activated by the next even number
+        LEDmidiSignal = 48 * 2,
+        activeLED = MC7000.isVinylMode[deckNumber] ? Math.round(position * LEDmidiSignal) % LEDmidiSignal : value * LEDmidiSignal;
```

Comment:
Can you explain what difference it makes whether the controller is in vinyl mode?

Reply:
If you are in Vinyl mode then the Platter LEDs follow the spinnies defined with default 33.3 rpm. If you switch off Vinylmode then the position of the track from start to end is shown representing a full circle.

Reply:
Does the controller behave like that in the commercial mappings as well?

Reply:
I don't care - sorry - This is not a Serato clone. But to answer your question - I don't even know as I am on plain Linux and cannot test Serato but I believe they only show spinnies, regardless of the mode. One more thing: Serato would by default start with all LEDs switched on which a lot of people complained it is blinding. Shall I make this also default? No! I set the default all LEDs off and only illuminate 1 LED. If we make a mapping then for the best usage but not to clone what others do - right? I am glad you support me so much now so please do not understand me wrong. Thank you!

Reply:
From the Mixxx Wiki: 
>  If your controller was specifically designed for DJing and has labels on the controls, make your mapping do what the labels say. However, you do not need to exactly follow the labels or mappings the manufacturer made for other software. If you think there is a better way to map it or the manufacturer's mapping does not make sense with Mixxx (or just does not make sense), map it how you think it should be. You are encouraged to map additional features not included in the manufacturer's mappings, but not at the expense of excluding functionality the controller is labeled for.
>
> Focus your mapping on functionality that is useful to have easy, quick access to while mixing. It is okay to leave parts of Mixxx unmapped that are not changed frequently while mixing; you do not need to map everything. The user can still use their mouse and keyboard to access functions not mapped on the controller. 

Personally, I feel like when a mapping gets merged, it should act as you'd expect. As long as this special behavior is stated clearly on its wiki page, I'm totally fine with it.

Reply:
Perfect, the spinnies are not labelled and I added more functionality as needed. You don't need spinnies when not in scratch mode and it is explained inside the WIKI page... https://www.mixxx.org/wiki/doku.php/denon_mc7000#leds

Reply:
Then its fine.